File: frontend/src/types/events.js
// 事件名常量必须与 Go internal/core/events.go 一一对应。
// 改动这里之前请先检查后端。

export const EVENT_PROGRESS = 'task:progress'
export const EVENT_LOG = 'task:log'
export const EVENT_DONE = 'task:done'
export const EVENT_ERROR = 'task:error'

export const LOG_INFO = 'info'
export const LOG_WARN = 'warn'
export const LOG_ERROR = 'error'

// 输出策略
export const OUTPUT_PER_KEYWORD = 'per_keyword'
export const OUTPUT_MERGED = 'merged'
export const OUTPUT_PER_SOURCE = 'per_source'

// 拆分模式
export const SPLIT_BY_SHEET = 'by_sheet'
export const SPLIT_BY_ROWS = 'by_rows'
export const SPLIT_BY_COLUMN = 'by_column'
export const SPLIT_BY_KEYWORD = 'by_keyword'

File: internal/core/events.go
package core

// 事件名常量。前端 frontend/src/types/events.js 必须镜像这些常量，
// 禁止在任何地方硬编码字符串字面量。
const (
	EventProgress    = "task:progress"
	EventLog         = "task:log"
	EventDone        = "task:done"
	EventError       = "task:error"
	EventFileBlocked = "task:file-blocked"
)

// 日志级别常量。
const (
	LogInfo  = "info"
	LogWarn  = "warn"
	LogError = "error"
)
